### PR TITLE
Adding logs to the Meteor Gun

### PR DESCRIPTION
**Pull request by DeityLink**

### CODE REVIEW

**Comment by PJB3005** [comment]
Fixes in the description please.

**Comment by Imagebard** [comment]
#10104

**Comment by Imagebard** [comment]
Wait
Does this fix meteor pens AND guns, or just one of them?
If not, don't add "fixes X" to anything.

File: code/game/gamemodes/meteor/meteors.dm [hunk 1]
```diff
@@ -468,7 +468,9 @@ var/list/blob_candidates = list()
 
 	playsound(user, 'sound/weapons/rocket.ogg', 100)
 
-	new projectile_type(get_turf(src), get_turf(A))
+	var/obj/item/projectile/meteor/in_chamber = new projectile_type(get_turf(src), get_turf(A))
+
+	log_attack("[user.name] ([user.ckey]) fired \the [src] (proj:[in_chamber.name]) at [A] [ismob(A) ? "([A:ckey])" : ""] ([A.x],[A.y],[A.z])" )
```
Comment (ComicIronic) [inline code/game/gamemodes/meteor/meteors.dm]: It is better to use add_logs() for this - keeps the formatting consistent and stops you using COLONS

Comment (DeityLink) [inline code/game/gamemodes/meteor/meteors.dm]: I'd rather use the same formatting used by every other guns.

**Comment by DeityLink** [comment]
@Imagebard the old meteor gun/pen are an entirely different item.

**Comment by Imagebard** [comment]
Is that still in the code?

**Comment by DeityLink** [comment]
Yeah the old ones are still in the code. I assumed whoever made them had their reasons so I didn't remove them. That new meteor gun is intended exclusively as an admin/debug tool, since you can set any type of meteor as projectile, including blob cores.

File: code/game/gamemodes/meteor/meteors.dm (hunk 1 above)
Comment (ComicIronic) [inline code/game/gamemodes/meteor/meteors.dm]: And I'd rather the meteor gun was actually a gun, but it isn't, and add_logs is useful for doing everything needed for proper logging, including LAssailant, which you don't do here.

So, add_logs.

**Comment by Imagebard** [comment]
That exact thing needs logging.
Some admin fired a meteor in the lobby, killing a lot of players that were still generating, and nobody knew WHO DID IT.

**Comment by CptWad** [comment]
meteor pens are a meme from the original goon release when a bug existed where you could shoot meteors out of pens

**Comment by Shadowmech88** [comment]
> when a bug existed where you could shoot meteors out of pens

I can't imagine how they could have managed to write code that could unintentionally cause this.
It's just so far out of left field.

**Comment by ComicIronic** [comment]
Href, probably.

**Comment by CptWad** [comment]
Theres even a plaque about it
plaqe

**Comment by clusterfack** [comment]
Wait what the fuck Deity, why is this not an actual gun.

**Comment by DeityLink** [comment]
Because it's only a "gun" in the name. It has nothing "gun-like" beside the sprite. It's just a tool that spawns meteors. Guns have lots of extra procs and variables that would conflict with the use meant for that tool.

**Comment by Imagebard** [comment]
>It has nothing "gun-like" beside the sprite

We're talking about meteor pens here right
Anyways I don't care as long as meteor explosions are logged.

**Comment by clusterfack** [comment]
Okay so its an item.

And it fires projectiles.

What you are describing is a gun deity.

**Comment by clusterfack** [comment]
If it was a gun we wouldn't be having this discussion.

**Comment by Imagebard** [comment]
Meteors aren't projectiles anymore, are they.
I remember that this was changed so meteors could hit people that are lying down.

**Comment by clusterfack** [comment]
How long ago was that? They were converted to projectiles not long ago.

**Comment by Imagebard** [comment]
>They were converted to projectiles not long ago.

Okay lets ask @dylanstrategie about this one

**Comment by clusterfack** [comment]
Ask him what about it?

**Comment by Imagebard** [comment]
Or its possible to just check the code, now that I think about it.
It seemed like the logical conclusion since he has been working on the meteor gamemode.

**Comment by ghost** [comment]
I just made meteors projectiles and deleted the old meteor projectile, which was an actual thing and shit

Didn't modify the gun, although I did give it a small test run and it was fun. Shame it's not logged I guess

**Comment by DeityLink** [comment]
We're not even talking about the same object.

**Comment by ghost** [comment]
I'm answering the person who pinged me

**Comment by Imagebard** [comment]
>Didn't modify the gun, 

So meteors are now projectiles?
Are you telling me that Meteor Pens/Guns could be converted to be a child of guns?

**Comment by DeityLink** [comment]
The old meteor gun/pen (made by dylan) are already child of guns. They have gun cooldown and all that, and you cannot set the type of projectile.

**Comment by ghost** [comment]
I didn't make the guns at all though. They have always been there and fired a snowflake "meteor projectile" that was unlinked from my meteor rework

When I switched meteors to become projectiles, I simply dumped the old snowflake overboard and made the gun fire real, reworked meteors. It's working fine, as tested, but is missing logging for whatever reasons

**Comment by DeityLink** [comment]
The gun that's missing logging is NOT that gun, dylan. It's a tool I made a few weeks ago, that I dubbed "meteor gun", so that I could test the new blob meteors.

Its existence is purely to test and debug meteor related shit, but once I spawned one in the middle of the start area and found out the hard way that players can grab items there. Someone instantly grabbed and fired it, killing everyone in the lobby.

**Comment by MercuryDoll** [comment]
So does that mean we have... three meteor guns in the code now?

**Comment by DeityLink** [comment]
The former two (including the pen) are fit for adminbus and distributed to players when causing mayhem is the objective. The newer one is intended for admin/coder use exclusively (outside of post-round end) due to being able to fire much faster and spawn any type of meteor including the bigger ones.

**Comment by clusterfack** [comment]
There's still literally zero reason it shouldn't be gun

**Comment by DeityLink** [comment]
And what's your reason beside "it spawns atoms that are children of projectile" ?

**Comment by Shadowmech88** [comment]
Why do you need a reason besides that?
After all, there are zero reasons that it _shouldn't_ be a gun.
1 > 0

**Comment by NigglyWiggly** [comment]
Why not just make the regular old meteor guns that don't fire normal meteors into the kind that do?

Oh wait this is about your different meteor gun that fires actual meteors.

Why the fuck do we have so many meteor guns

**Comment by clusterfack** [comment]
Oh I don't know, maybe so we don't run into problems in the future with random items we made ourselves not having logs for shooting projectiles? This VERY issue.

**Comment by ComicIronic** [comment]
> you click to shoot
> it spawns a projectile
> it generates attack logs

**Comment by DeityLink** [comment]
@ComicIronic guns make a lot of extra checks that'd slow down the spawn rate.

**Comment by clusterfack** [comment]
If those checks are hardcoded into the gun level they should be more configurable

**Comment by ComicIronic** [comment]
They are configurable: you can disable pretty much every firing check guns make using bool vars.

**Comment by ComicIronic** [comment]
I don't know what's going to happen to this.

We are in the position where however many meteor guns we have, they are all able to become gun-code guns, which would invalidate this PR. The only change I wanted still hasn't been made, so I'm not going to merge it.

**Comment by DeityLink** [comment]
@ComicIronic Your change has been made.

**Pull request merged**